### PR TITLE
Update README using the one from the enarx.github.io repo

### LINKED ISSUE #137: Update enarx/enarx README to better reflect the project
The `enarx/enarx` README currently reads "This is a placeholder repository." We do not want this to be the first thing newcomers see when they look at our premiere repo. We should have something better there, even if it isn't an exhaustive description of the project. A link to the wiki would be a good idea.

### PR DESCRIPTION
**Pull request by axelsimon**
Update README, using the one initially created for the enarx.github.io repo.
Closes #137.

There's also a PR to update the README in the other repo, enarx/enarx.github.io#25.

### CODE REVIEW

**Comment by axelsimon** [comment]
Closes #137, btw.

File: README.md [hunk 1]
```diff
@@ -1,6 +1,44 @@
 # Enarx
-A client for deploying WebAssembly applications into Enarx Keeps.
 
-This is currently a placeholder repository.
+## Introduction
+Enarx is an application deployment system enabling applications to run within Trusted Execution Environments (TEEs) without rewriting for particular platforms or SDKs. It handles attestation and delivery into a run-time “Keep” based on WebAssembly, offering developers a wide range of language choices for implementation. Enarx is CPU-architecture independent, enabling the same application code to be deployed across multiple targets, abstracting issues such as cross-compilation and differing attestation mechanisms between hardware vendors. Work is currently underway on AMD SEV and Intel SGX.
+
+We've known for a long time that we need encryption for data at rest and in transit: Enarx helps you do encryption for data in use.
+
+[Read more](https://github.com/enarx/enarx.github.io/wiki/Enarx-Introduction)
```
[the remark this answers is not in the document]
Comment (axelsimon) [reply README.md]: thanks, well spotted, will amend

File: README.md [hunk 2]
```diff
@@ -1,6 +1,44 @@
...
+   * multiple silicon architectures; 
+   * multiple server types;
+   * public cloud or private cloud;
+ * employ serverless deployment and execution mechanisms.
+
+The mechanism itself should be capable of being FIPS-certified.
+
+## Getting Started
+
+For more information, please try our [wiki](https://github.com/enarx/enarx/wiki), which includes further details on how to get more information.  It's also where we plan to keep up-to-date project information.
```
[the remark this answers is not in the document]
Comment (axelsimon) [reply README.md]: The logic so far is have people land on the README, give them a bit of information, then refer them to the full documentation for more.

I agree there are more decisions to be made regarding the wiki's home vs. the wiki's introduction page. But this PR intended to just resolve the initial issue of having an outdated README on enarx/enarx.

Does that resolve your concerns?

[the remark this answers is not in the document]
Comment (mbestavros) [reply README.md]: Just something I noted. It's not a blocker in any case.

**Comment by axelsimon** [comment]
Rebased and force pushed, should be ok now. Waiting on the CI tests to complete.

On this subject: running a whole Travis CI build for a change in an .md file does seem like wasted computing resources, i wonder if there's something we could do about that and reduce our energy footprint.

File: README.md [hunk 3]
```diff
@@ -1,6 +1,44 @@
 # Enarx
-A client for deploying WebAssembly applications into Enarx Keeps.
 
-This is currently a placeholder repository.
+## Introduction
+Enarx is an application deployment system enabling applications to run within Trusted Execution Environments (TEEs) without rewriting for particular platforms or SDKs. It handles attestation and delivery into a run-time “Keep” based on WebAssembly, offering developers a wide range of language choices for implementation. Enarx is CPU-architecture independent, enabling the same application code to be deployed across multiple targets, abstracting issues such as cross-compilation and differing attestation mechanisms between hardware vendors. Work is currently underway on AMD SEV and Intel SGX.
+
+We've known for a long time that we need encryption for data at rest and in transit: Enarx helps you do encryption for data in use.
```
Comment (steveeJ) [inline README.md]:
```suggestion
We've known for a long time that we need encryption for data at rest and in transit: Enarx helps you do encryption for data as well as algorithms in use.
```

File: README.md [hunk 4]
```diff
@@ -1,6 +1,44 @@
...
+Create a way to create and run "private, fungible, serverless" applications using Trusted Execution Environments (TEEs). In other words, to provide a platform abstraction for TEEs.
+
+They should:
+
+ * be confidential from the host and other workloads;
+ * be able to run on
+   * multiple silicon architectures; 
+   * multiple server types;
+   * public cloud or private cloud;
+ * employ serverless deployment and execution mechanisms.
```
Comment (steveeJ) [inline README.md]: Rather than the buzzword I'd describe the properties this implies in your understanding.

File: README.md [hunk 5]
```diff
@@ -1,6 +1,44 @@
...
+
+Building the various components of Enarx is currently complex: we are working on this.  Please contact us for help.
+
+(2020-01-01) It is also worth mentioning that it isn't yet possible to run Enarx.  We're working hard on it and we'd love people to work with us. We hope to be adding more information very soon, to allow you to get started.
+
+## Authors
+
+* **Mike Bursell** - *Initial work* - [MikeCamel](https://github.com/MikeCamel)
+
+See also the list of [people](https://github.com/orgs/enarx/people) who participated in this project.
```
Comment (steveeJ) [inline README.md]: I'd use present tense here. The participation is ongoing :-)

File: README.md (hunk 3 above)
Comment (npmccallum) [inline README.md]: I'd suggest being a little less colloquial here. Perhaps:

```suggestion
Enarx aims to transparently enable encryption for code and data:
  * in transit
  * at rest
  * in use
```

File: README.md (hunk 5 above)
Comment (MikeCamel) [reply README.md]: Indeed.  please fix.

File: README.md (hunk 4 above)
Comment (axelsimon) [reply README.md]: This is a good point @steveeJ, however i think that will require to put thought into it than what this issue and PR intended to do. I'll resolve for now but open an issue to track this.

File: README.md (hunk 3 above)
Comment (axelsimon) [reply README.md]: Good point. Will incorporate.

Comment (axelsimon) [reply README.md]: I don't think i agree here: i think we impart less context and don't make our intentions with Enarx as clear by just saying it aims to transparently enable encryption for code and data. Arguably, LUKS already does just that. Sure, it's not for code or data in use.
Which is why i think the context of encrypting data at rest, data in transit and data in use is helpful.

Adding "code and data" is useful however, and i've incorporated Stefan's suggestion.

File: README.md (hunk 5 above)
Comment (axelsimon) [reply README.md]: Fixed in latest pushed branch.

**Pull request merged**